pos_payment_terminal/static/src/js/payment_terminal.js

```diff
@@ -120,21 +141,30 @@ odoo.define("pos_payment_terminal.payment", function(require) {
                                     transaction.transaction_id ===
                                     pay_line.terminal_transaction_id
                                 ) {
+                                    got_response = true;
                                     // Look for the transaction corresponding to
                                     // the payment line.
                                     this._oca_update_payment_line_terminal_transaction_status(
                                         pay_line,
                                         transaction
                                     );
                                     if (
-                                        pay_line.terminal_transaction_success !== null
+                                        pay_line.terminal_transaction_success !== null &&
+                                        pay_line.terminal_transaction_success !== undefined
                                     ) {
                                         resolve(pay_line.terminal_transaction_success);
                                         // Stop the loop
                                         clearInterval(timerId);
                                     }
                                 }
                             }
+                            if (!got_response && retries > 30) {
+                                // if after 30 seconds still no word about the status of the transaction,
+                                // then consider it lost in space. Let user decide the status.
```

Comment:
I'm not sure if this plays well with other drivers such as Telium, which may not return any status until there is a definite Approved or Cancelled - in that case 30 seconds may be too soon. Maybe this should be made configurable?

Reply:
Or we could just remove this, since the 'waiting for card' screen does show a 'Cancel' button, so the user is not stuck.